legacy/jacorb/src/main/java/org/jboss/as/jacorb/JacORBExtension.java

```diff
@@ -185,11 +185,83 @@ protected static void registerTransformers_1_3(final SubsystemRegistration subsy
+                JacORBSubsystemDefinitions.ORB_CONN_OUTBUF_CACHE_TIMEOUT);
+        builder.getAttributeBuilder().setValueConverter(
+                new ChangeUndefinedDefaultValueConverter(new ModelNode().set(Integer.MAX_VALUE)),
+                JacORBSubsystemDefinitions.ORB_CONN_MAX_SERVER_CONNECTIONS);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(24)),
+                JacORBSubsystemDefinitions.ORB_CONN_MAX_MANAGED_BUF_SIZE);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(10)),
+                JacORBSubsystemDefinitions.POA_QUEUE_MIN);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(100)),
+                JacORBSubsystemDefinitions.POA_QUEUE_MAX);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(5)),
+                JacORBSubsystemDefinitions.POA_REQUEST_PROC_POOL_SIZE);
+        builder.getAttributeBuilder().setValueConverter(new ChangeUndefinedDefaultValueConverter(new ModelNode().set(32)),
+                JacORBSubsystemDefinitions.POA_REQUEST_PROC_MAX_THREADS);
+        builder.getAttributeBuilder().setValueConverter(
+                new ChangeDefaultValueConverter(new ModelNode(Constants.NAME_SERVICE_INIT_REF), new ModelNode(
+                        "JBoss/Naming/root")), JacORBSubsystemDefinitions.NAMING_ROOT_CONTEXT);
+    }
+
+    private static class ChangeUndefinedDefaultValueConverter extends AttributeConverter.DefaultAttributeConverter {
+
+        private final ModelNode defaultValue;
+
+        public ChangeUndefinedDefaultValueConverter(final ModelNode defaultValue) {
+            this.defaultValue = defaultValue;
+        }
+
+        @Override
+        protected void convertAttribute(final PathAddress address, final String attributeName, final ModelNode attributeValue,
+                final TransformationContext context) {
+            if (!attributeValue.isDefined()) {
+                attributeValue.set(defaultValue);
+            }
+        }
+    }
+
+    private static class ChangeDefaultValueConverter extends AttributeConverter.DefaultAttributeConverter {
+
+        private final ModelNode newDefaultValue;
+        private final ModelNode oldDefaultValue;
+
+        public ChangeDefaultValueConverter(final ModelNode newDefaultValue, final ModelNode oldDefaultValue) {
+            this.newDefaultValue =newDefaultValue;
+            this.oldDefaultValue = oldDefaultValue;
+        }
+
+        @Override
+        protected void convertAttribute(final PathAddress address, final String attributeName, final ModelNode attributeValue,
+                                        final TransformationContext context) {
+            if (!attributeValue.equals(newDefaultValue)) {
```

Comment:
I don't think the "!" belongs here.

I don't think this conversion is right though. AIUI the attributeValue param should be what the user configured, likely undefined. If they specifically configured "NameService" I don't see why we would change it to something else. If they didn't configure it at all, that's where changing the value comes in, to make sure the legacy server config matches the current server config.

In this case though, is it correct to change the default value? Won't apps expect the JNDI binding to be what it used to be? Can this be handled in the OSH; i.e. convert undefined to "JBoss/Naming/root"? The iiop-jdk ORB should be able to handle that setting, right? Adding a ParameterCorrector to JacORBSubsystemDefinitions.NAMING_ROOT_CONTEXT can do the conversion of undefined to "JBoss/Naming/root".